src/main/java/org/testng/ITestRunnerFactory2.java

```diff
@@ -10,6 +10,7 @@
  * A factory for TestRunners to be used by SuiteRunners.
  */
 public interface ITestRunnerFactory2 extends ITestRunnerFactory {
```

Comment:
Please have this interface contents moved to its parent and deprecate this interface.

Reply:
+1, where the default impl will just call the other newTestRunner method.

Reply:
@juherr so which one should be mandatory / default?.. with `Map` or without?..

Reply:
Only the original from ITestRunnerFactory can be mandatory. Default is for new methods like ITestRunnerFactory2. But we can deprecate the original method at the same time if it makes sense and always calls the new method in its usage in testng core.